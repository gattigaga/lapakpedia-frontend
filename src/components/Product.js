import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

import { toCurrency, cutText } from "helpers/formatter";

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const Container = styled.div`
  border: 1px solid #eee;
  cursor: pointer;
  height: 100%;
  position: relative;
`;

const Name = styled.h5`
  font-family: Roboto;
  color: black;
  margin: 0px;
  text-decoration: none;
`;

const Category = styled.p`
  font-family: Roboto;
  font-size: 10px;
  color: #aaa;
  margin-top: 4px;
  margin-bottom: 14px;
`;

const Price = styled.p`
  font-family: Roboto;
  font-size: 16px;
  margin: 0px;
  color: #ff8800;
`;

const Image = styled.img`
  width: 100%;
  height: 160px;
  object-fit: cover;
`;

const DetailContainer = styled.div`
  padding: 12px;
  box-sizing: border-box;
`;

const Layer = styled.div`
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
  display: flex;
`;

const WarnText = styled.p`
  font-family: Roboto;
  font-size: 24px;
  color: white;
  margin: auto;
`;

const Product = ({ image, name, category, price, href, isNoStock }) => {
  const totalWords = name.split(" ").length;

  return (
    <StyledLink to={href}>
      <Container>
        <Image src={image} />
        <DetailContainer>
          <Name>{totalWords > 7 ? cutText(name) : name}</Name>
          <Category>{category}</Category>
          <Price>$ {toCurrency(price)}</Price>
        </DetailContainer>
        {isNoStock && (
          <Layer>
            <WarnText>Out of Stock</WarnText>
          </Layer>
        )}
      </Container>
    </StyledLink>
  );
};

Product.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  category: PropTypes.string,
  price: PropTypes.number,
  href: PropTypes.string,
  isNoStock: PropTypes.bool
};

Product.defaultProps = {
  href: "#"
};

export default Product;
