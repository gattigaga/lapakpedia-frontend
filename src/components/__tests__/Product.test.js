import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Product from "components/Product";

describe("Product", () => {
  const setup = propOverrides => {
    const props = {
      ...propOverrides,
      name: "MacBook Pro",
      category: "Technology",
      price: 2800,
      href: "https://github.com"
    };

    const wrapper = shallow(<Product {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders in no stock state", () => {
    const { wrapper } = setup({ isNoStock: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
