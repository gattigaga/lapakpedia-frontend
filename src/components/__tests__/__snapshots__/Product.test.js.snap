// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should renders default 1`] = `
<Styled(Link)
  to="https://github.com"
>
  <styled.div>
    <styled.img />
    <styled.div>
      <styled.h5>
        MacBook Pro
      </styled.h5>
      <styled.p>
        Technology
      </styled.p>
      <styled.p>
        $ 
        2,800
      </styled.p>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;

exports[`Product should renders in no stock state 1`] = `
<Styled(Link)
  to="https://github.com"
>
  <styled.div>
    <styled.img />
    <styled.div>
      <styled.h5>
        MacBook Pro
      </styled.h5>
      <styled.p>
        Technology
      </styled.p>
      <styled.p>
        $ 
        2,800
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.p>
        Out of Stock
      </styled.p>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
